app/scripts/directives/editEnvironmentFrom.js

```diff
@@ -50,11 +50,12 @@
       ctrl.editEnvironmentFromForm.$setDirty();
     };
 
+    ctrl.isEnvFromSelectable = function() {
+      return !_.isEmpty(ctrl.envFromSelectorOptions);
+    };
+
     ctrl.isEnvFromReadonly = function(entry) {
-      return ctrl.isReadonlyAny ||
-        entry.isReadonlyValue === true ||
-        ((entry.secretRef || entry.configMapRef) && !entry.selectedEnvFrom) ||
-        _.isEmpty(ctrl.envFromSelectorOptions);
+      return ctrl.isReadonlyAny || entry.isReadonlyValue === true || ((entry.secretRef || entry.configMapRef) && !entry.selectedEnvFrom);
```

Comment:
We could simplify this a lot by just saying the whole thing is either read-only or it isn't. We currently don't have a need to make individual values read-only.

Reply:
That works, removing the "isReadonlyAny" and tweaking some of the related logic